refactor(fivetran_sdk): Restructure the fivetran_sdk repo (#88)
fivetran/fivetran_sdk: development-guide.md

* Partners should not add the proto files to their repos. Proto files should be pulled in from this repo at build time and added to `.gitignore` so they are excluded.
* Always use proto files from latest release and update you're code if necessary. Older releases proto files can be considered deprecated and will be expired at later date.

> * New partners should use [V2 protos](v2), while only old partners should continue using [V1 protos](v1).

Reviewer: @5tran-alexil please suggest proper wording for this.

Reviewer 2: A suggestion from me
"New partners should use the latest proto files  [V2 protos](v2), older files [V1 protos](v1) are provided for reference during migration and will be removed in the future."

Author: @alsnkline
"> NOTE: If you're a new partner that just started working with the Connector SDK, use [V2 protos](v2). We have provided [V1 protos](v1) for partners that have used them to build their SDK connectors, but these will be removed once the migration to V2 is complete."
What about this?

Author: Updated it.

> * New partners should use [V2 protos](v2), while only old partners should continue using [V1 protos](v1).

Reviewer 3: ```suggestion
* New partners that just started working with the Connector SDK should use [V2 protos](v2). Only partners that have already used [V1 protos](v1) to build SDK connectors should continue using them. 
```

- Partner code should capture and relay a clear message when the account permissions are not sufficient.

### User alerts

> > Note: Available in V2 only

Reviewer 3: ```suggestion
> NOTE: Available in V2 only.
```

> - Text Field: A standard text input field for user text entry. You can provide a `title` displayed above the field. You can indicate whether the field is `required`, and you may also include an optional `description` displayed below the field to help explain what the user should complete.

Reviewer 3: ```suggestion
- Text field: A standard text input field for user text entry. You can provide a `title` displayed above the field. You can indicate whether the field is `required`, and you may also include an optional `description` displayed below the field to help explain what the user should complete.
```

- Dropdown: A drop-down menu that allows users to choose one option from the list you provided.

> - Toggle Field: A toggle switch for binary options (e.g., on/off or yes/no).

Reviewer 3: ```suggestion
- Toggle field: A toggle switch for binary options (e.g., on/off or yes/no).
```

> - Conditional Fields (Available in V2): This feature allows you to define fields that are dependent on the value of a specific parent field. The message consists of two nested-messages: `VisibilityCondition` and a list of dependent form fields. The `VisibilityCondition` message specifies the parent field and its condition value. The list of dependent fields defines the fields that are shown when the value of the parent field provided in the setup form matches the specified condition field.

Reviewer 3: ```suggestion
- Conditional fields (Available in V2): This feature allows you to define fields that are dependent on the value of a specific parent field. The message consists of two nested-messages: `VisibilityCondition` and a list of dependent form fields. The `VisibilityCondition` message specifies the parent field and its condition value. The list of dependent fields defines the fields that are shown when the value of the parent field provided in the setup form matches the specified condition field.
```

> ## Source Connector guidelines

Reviewer: under System Columns section, we should add History mode columns or add link to refer to History mode section to know about additional columns specific to history mode.

Author: I have added info about it and linked it to history mode documentation.

    - `unmodified_string` value is used to indicate columns in `update_files` where the values did not change.

#### WriteHistoryBatchRequest

> > Note: This rpc call is available in V2

Reviewer 3: ```suggestion
> NOTE: This RPC call is available in V2 only.
```
